app/Console/Commands/Testing/TestAdvisorGeneration.php

```
@@ -0,0 +1,187 @@
+            
+            // Test variable extraction
+            $variables = $templateService->extractVariables($template);
+            $this->info('✓ Found ' . count($variables) . ' variables: ' . implode(', ', array_slice($variables, 0, 5)) . '...');
+            
+            // Test variable substitution
+            $testVars = [
+                'advisor_name' => 'Test Advisor',
+                'personality_trait' => 'analytical and thoughtful',
+                'speaking_style' => 'clear and concise'
+            ];
+            $processed = $templateService->substituteVariables($template, $testVars);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$processed'. (Unused Code Rules)

(UnusedLocalVariable)